src/Halogen/Portal.purs

```diff
@@ -44,46 +52,79 @@ type State query input output
 -- | ```purs
 -- | -- if `Nothing` is provided as the target HTMLElement, then the `<body>`
 -- | -- tag will be used
--- | HH.div_ 
--- |   [ portal _modal unit Modal.component modalInput (Just element) handler ]
+-- | HH.div_
+-- |   [ portal ntIdentity _modal unit Modal.component modalInput (Just element) handler ]
 -- |
--- | -- for comparison, this is how you would mount the component _not_ via 
+-- | -- for comparison, this is how you would mount the component _not_ via
 -- | -- a portal
 -- | HH.div_
 -- |   [ HH.slot _modal unit Modal.component modalInput handler ]
 -- | ```
 portal ::
-  forall query action input output slots label slot _1.
+  forall query action input output slots label slot _1 m n.
   Row.Cons label (H.Slot query output slot) _1 slots =>
   IsSymbol label =>
   Ord slot =>
+  MonadAff m =>
+  m (NT n Aff) ->
   SProxy label ->
   slot ->
-  H.Component HH.HTML query input output Aff ->
+  H.Component HH.HTML query input output n ->
   input ->
   Maybe HTMLElement ->
   (output -> Maybe action) ->
-  H.ComponentHTML action slots Aff
-portal label slot childComponent childInput htmlElement handler =
+  H.ComponentHTML action slots m
+portal contextualize label slot childComponent childInput htmlElement handler =
   handler
-    # HH.slot label slot component
+    # HH.slot label slot (component contextualize)
         { child: childComponent
         , input: childInput
         , targetElement: htmlElement
         }
 
+-- | Run a portal component that is already in `Aff`.
```

Comment:
Could this be relaxed to `MonadAff`?

Reply:
Hmm. The input `H.Component HH.HTML query input output Aff` has to be `Aff`, of course, but you're right that the output can be `MonadAff`.